src/CoreWCF.Kafka/src/CoreWCF/Channels/KafkaTransportBindingElement.cs

```diff
@@ -0,0 +1,108 @@
+        private KafkaDeliverySemantics _deliverySemantics;
+        private KafkaErrorHandlingStrategy _errorHandlingStrategy = KafkaErrorHandlingStrategy.Ignore;
+        internal ConsumerConfig Config { get; } = new();
+
+        public KafkaDeliverySemantics DeliverySemantics
+        {
+            get => _deliverySemantics;
+            set
+            {
+                if (!KafkaDeliverySemanticsHelper.IsDefined(value))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+
+                _deliverySemantics = value;
+            }
+        }
+
+        public KafkaMessageEncoding MessageEncoding
+        {
+            get => _messageEncoding;
+            set
+            {
+                if (!KafkaMessageEncodingHelper.IsDefined(value))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+
+                _messageEncoding = value;
+            }
+        }
+
+        public KafkaErrorHandlingStrategy ErrorHandlingStrategy
+        {
+            get => _errorHandlingStrategy;
+            set
+            {
+                if (!KafkaErrorHandlingStrategyHelper.IsDefined(value))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+
+                _errorHandlingStrategy = value;
+            }
+        }
+
+        public string DeadLetterQueueTopic { get; set; }
+
+        public KafkaTransportBindingElement()
+        {
+        }
+
+        private KafkaTransportBindingElement(KafkaTransportBindingElement other)
+            : this()
+        {
+            _deliverySemantics = other.DeliverySemantics;
+            _messageEncoding = other.MessageEncoding;
+            Config = other.Config;
+        }
+
+        public override BindingElement Clone()
+        {
+            return new KafkaTransportBindingElement(this);
+        }
+
+        public override string Scheme => "net.kafka";
+
+        public override T GetProperty<T>(BindingContext context)
+        {
+            if (context == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(context));
+            }
+
+            if (typeof(T) == typeof(ISecurityCapabilities))
+            {
+                return null;
+            }
+
+            return base.GetProperty<T>(context);
+        }
+
+        public override QueueTransportPump BuildQueueTransportPump(BindingContext context)
+        {
+            var serviceProvider = context.BindingParameters.Find<IServiceProvider>();
+            var serviceDispatcher = context.BindingParameters.Find<IServiceDispatcher>();
+            var logger = serviceProvider.GetRequiredService<ILogger<KafkaTransportPump>>();
+
+            return new KafkaTransportPump(this, logger, serviceDispatcher, DeliverySemantics);
+        }
```

Comment:
I'm not saying this is how you should do things, especially as only one parameter comes from DI, but an alternative would be to add KafkaTransportPump to DI then use `ActivatorUtilities.CreateInstance(serviceProvider, typeof(KafkaTransportPump), serviceDispatcher, DeliverySemantics)`.

Reply:
Thanks @mconnew  i did not know ActivatorUtilities.CreateInstance (i however know Activator.CreateInstance).
Having a look to what is done for RabbitMQ it seems that RabbitMqTransportPump is added to DI through an extension method named AddServiceModelRabbitMqSupport which is never called and RabbitMqTransportBindingElement is manually newing the pump instance which mean that the service registration (and thus the extension method) is useless.
I think we should align both implementations what is your view about this ?

Reply:
I personally like the extension method as this explicitly signifies you are using RabbitMQ... I will look at the RabbitMQ piece.. thanks for bring it up

Reply:
@birojnayak please see https://github.com/CoreWCF/CoreWCF/pull/1127